autoware_launch/config/map/lanelet2_map_loader.param.yaml

```diff
@@ -1,5 +1,7 @@
 /**:
   ros__parameters:
+    enabled_dynamic_lanelet_loading: true
```

Comment:
```suggestion
    enable_differential_map_loading: true
```
`enable_differential_map_loading` is correct?

Reply:
Even fixing this did not work for me and I needed to add the launch argument `enable_differential_map_loading:=true`, otherwise the feature was not enabled.

Reply:
@StepTurtle 
Thank you for your response.

> Would you like me to remove the parameter from either the config file or the parameter file to avoid this issue?

Other packages also have the same parameter in two different locations. So it's fine as it is.

Reply:
@StepTurtle Sorry, I misunderstood.
Please remove 
[autoware_launch/launch/components/tier4_map_component.launch.xml](https://github.com/autowarefoundation/autoware_launch/pull/872/files#diff-9c70a1db8613ccb66bc7f366e315010c0061130e28bd1affbfb3a17c73deede8) >>> line 3.

By the way, is the `enable_differential_map_loading` parameter still necessary? Since it now works with a single map, it seems like `enable_differential_map_loading` could always be set to true without any issues. If that's the case, I think we can remove the need to toggle `enable_differential_map_loading` and delete it. What do you think?

Reply:
The main reason I do this within the launch file is control following three part in `launch/tier4_map_launch/launch/map.launch.xml`. If we remove this, we have to run extra one more `lanelet2_map_visualization` and `dynamic_lanelet_provider` on default. I guess we need this parameter on launch file.

```
<arg name="use_multithread" default="false" unless="$(var enable_differential_map_loading)"/>
<arg name="use_multithread" default="true" if="$(var enable_differential_map_loading)"/>
```

```
<composable_node
        pkg="autoware_dynamic_lanelet_provider"
        plugin="autoware::dynamic_lanelet_provider::DynamicLaneletProviderNode"
        name="dynamic_lanelet_provider"
        if="$(var enable_differential_map_loading)"
      >
        <param from="$(var dynamic_lanelet_provider_param_path)"/>
        <remap from="input/odometry" to="/localization/kinematic_state"/>
        <remap from="output/lanelet2_map" to="dynamic_vector_map"/>
        <remap from="service/get_differential_lanelet_map" to="/map/get_differential_lanelet_map"/>
        <extra_arg name="use_intra_process_comms" value="$(var use_intra_process)"/>
        <extra_arg name="use_intra_process_comms" value="$(var use_intra_process)"/>
      </composable_node>

```

```
<composable_node pkg="map_loader" plugin="Lanelet2MapVisualizationNode" name="dynamic_lanelet2_map_visualization" if="$(var enable_differential_map_loading)">
        <remap from="input/lanelet2_map" to="/map/dynamic_vector_map"/>
        <remap from="output/lanelet2_map_marker" to="/map/dynamic_vector_map_marker"/>
        <extra_arg name="use_intra_process_comms" value="$(var use_intra_process)"/>
</composable_node>
```

Reply:
> we have to run extra one more lanelet2_map_visualization and dynamic_lanelet_provider on default.

I believe that differential_map_loading will become mainstream for lanelet2 in the future, so it should be enabled by default.

Reply:
@yukkysaito @mitsudome-r Do you have any opinions on this? We are discussing whether to always enable differential map loading or to provide an option to toggle it with a flag.

Reply:
@yukkysaito @mitsudome-r  Friendly Reminder

Reply:
I think it would be fine to have it as an option. I agree with keeping it enabled by default.

Reply:
@yukkysaito Thank you!

@StepTurtle 

> I agree with keeping it enabled by default.


So, please modify the launch.xml file so that the default is set to true.
[autoware_launch/launch/components/tier4_map_component.launch.xml](https://github.com/autowarefoundation/autoware_launch/pull/872/files#diff-9c70a1db8613ccb66bc7f366e315010c0061130e28bd1affbfb3a17c73deede8)
```
  <arg name="enable_differential_map_loading" default="true"/>
```

And, modify the yaml file to avoid explicitly stating true/false.
[autoware_launch/config/map/lanelet2_map_loader.param.yaml](https://github.com/autowarefoundation/autoware_launch/pull/872/files#diff-1c852a89e3e0f70598f9efa7865cd45fa915a348fe20bfbf251eb38aae00afe3)
```
  enable_differential_map_loading: $(var enable_differential_map_loading)
```

Also modify [map/map_loader/config/lanelet2_map_loader.param.yaml](https://github.com/autowarefoundation/autoware.universe/pull/6241/files#diff-323c696c799e6df78f59a4e07c83ff4e94ec4aca3e8bb2cd7baf3a8c47313fb2) and [map/map_loader/schema/lanelet2_map_loader.schema.json](https://github.com/autowarefoundation/autoware.universe/pull/6241/files#diff-a59c82026e217ef26e29e1957ddf16f1f5ec4efdd3c51c85d621fa20b227595d)

Reply:
Changed the default value to `true`.

related commits:
- https://github.com/autowarefoundation/autoware.universe/pull/6241/commits/beb7fafd1cd88b4d4a5c573b6789e7f106b79368
- https://github.com/autowarefoundation/autoware_launch/pull/872/commits/4f145e2cf32b72253b9f931d5d710daf055c5cb4

Reply:
Update: Changed the parameter type from `boolean` to `string` because of the error form `json-schema-check`:

![Screenshot from 2024-09-03 15-17-50](https://github.com/user-attachments/assets/e34e286b-58a2-431c-a317-ca3ac803a03d)

related commit: https://github.com/autowarefoundation/autoware.universe/pull/6241/commits/af9ec44640a98983cbff55a679297cf7faf32839

Reply:
@StepTurtle Thank you!